### PR TITLE
Add regex format restriction for container_name

### LINKED ISSUE #34: Naming limitations for `container_name`
If I recall correctly, the `container_name` field is more restrictive than just _any_ UTF-8-encoded string. That should be specified in the spec.

### PR DESCRIPTION
**Pull request by hangyan**
Signed-off-by: Hang Yan <hangyan@alauda.io>

**What this PR does / why we need it**:
Add regex name restriction fort container_name feld.

**Which issue(s) this PR fixes**:

Fixes #34

Seems this regex restrictions does not appear in the runtime spec, and it's reported by this issue 
https://github.com/moby/moby/issues/3138. Accounding the comments, this regex also apply for users and images, so i think it's fair enough to add this restirctions. As for kubernetes, it can easily use some transform methods to fit in this restriction.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -316,6 +316,8 @@ container_name: my-web-container
 Compose implementation MUST NOT scale a service beyond 1 container if the Compose file specifies a
 `container_name`. Attempting to do so MUST result in an error.
 
+If represents, `container_name` SHOULD follow the regex fomrat of `[a-zA-Z0-9][a-zA-Z0-9_.-]+`
```
Comment (ndeloof) [inline spec.md:319]: typo

Comment (hangyan) [reply spec.md]: fixed

File: spec.md [hunk 2]
```diff
@@ -321,6 +321,8 @@ container_name: my-web-container
 Compose implementation MUST NOT scale a service beyond one container if the Compose file specifies a 
 `container_name`. Attempting to do so MUST result in an error.
 
+If represents, `container_name` SHOULD follow the regex fomrat of `[a-zA-Z0-9][a-zA-Z0-9_.-]+`
```
Comment (justincormack) [inline spec.md:324]: typo here "fomrat" not "format"

Comment (SuperSandro2000) [inline spec.md:324]:
```suggestion
If represents, `container_name` SHOULD follow the regex format of `[a-zA-Z0-9][a-zA-Z0-9_.-]+`
```

**Pull request merged**